static/scenes/railroad/js/present.js

```diff
@@ -0,0 +1,56 @@
+goog.provide('app.Present');
+
+class Present {
+
+    constructor(loader, scene, giftWrapMaterial) {
+        this.scene = scene;
+        this.inFlight = false;
+        this.landed = false;
+        this.pointIndex = 0;
+
+        const material0 = new THREE.MeshToonMaterial( {color: 0xF9D231}); 
+
+        loader.load( "models/gift.obj", obj => {
+            obj.scale.setScalar(0.001);
+            for (let i = 0; i < obj.children.length; i++) {          
+                if (i !== 4) {
+                  obj.children[i].material = material0;
+                } else {
+                  obj.children[i].material = giftWrapMaterial;
+                }
+              }
+            this.model = obj;
+            this.scene.getScene().add(obj);
+            console.log("Present added");
+        });
+    }
+
+    shoot(targetPosition) {
+        this.targetPosition = targetPosition;
+        this.startPosition = this.model.position;
+
+        const midpoint = new THREE.Vector3().add(this.targetPosition, this.startPosition).multiplyScalar(0.5);
+        midpoint.setY(midpoint.y + 3);
+        this.curve = new THREE.QuadraticBezierCurve3(this.startPosition, midpoint, this.targetPosition);
+        this.points = this.curve.getPoints(175);
```

Comment:
Instead of saving the points ahead of time and using the point index, we can do something like this:

- Create a variable for the total flight time
- Replace the point index variable with a variable that's the flight time so far
- In update, add the seconds to the flight time so far variable
- Then we can get a t value that will go from 0 to 1, where 0 is the start of the flight and 1 is the end by doing (flight time so far) / (total flight time).
- Using that, you can then use [Curve.getPoint](https://threejs.org/docs/#api/en/extras/core/Curve.getPoint) to get the point on the curve at that given time, and the copy the position from there

The advantages of doing it this way are:
- If the game runs on a slower computer such that there's a long time between rendering frames, the presents will still move at the same speed
- (minor) you don't need the extra memory of the points array.